File: valihahotel/reservations/src/main/java/com/valiha/reservation/core/interfaces/factory/RoomFactory.java
package com.valiha.reservation.core.interfaces.factory;

import com.valiha.reservation.core.entities.models.Category;
import com.valiha.reservation.core.entities.models.Room;

public interface RoomFactory {
  Room create(
    String id,
    String title,
    int price,
    String image,
    Category category
  );
}

File: website/src/core/entities/factory/RoomFactory.ts
import { Room } from "@/core/entities/models/Room";
import { RoomFactory } from "@/core/interfaces/factory/RoomFactory";
import { Category } from "../models/Category";

export class RoomFactoryImpl implements RoomFactory {
  create(
    id: string,
    title: string,
    price: number,
    image: string,
    category: Category
  ): Room {
    return new Room(id, title, price, image, category);
  }
}
